kafka/consumer/subscription_state.py

```diff
@@ -128,7 +158,11 @@ def change_subscription(self, topics):
 
         Raises:
             IllegalStateErrror: if assign_from_user has been used already
-            TypeError: if a non-str topic is given
+            TypeError: if a topic is
+                        - a non-str or
+                        - an empty string or
+                        - the topic name is '.' or '..' or
+                        - the topic name does not consist of ASCII-characters/'-'/'_'/'.'
```

Comment:
I would probably entirely remove this part of the docstring since the raising for invalid values now happens in `_ensure_valid_topic_name()`

Reply:
The internal method raises something. Shouldn't this be documented in the public method that uses this method (since it uses it only for that)? The checks are only encapsulated in a separate method to make it easier to reuse the checks later on.

Reply:
I would keep them in the docstring here, but update re ValueError